tests/datacon/neg/AutoliftedFields00.hs

```diff
@@ -0,0 +1,14 @@
+{-@ LIQUID "--exact-data-cons" @-}
+
+-- data decl in LH is missing and uses a LH-refined type alias incorrectly
+
+module AutoliftedFields00 where
+
+{-@ type Nat = { v : Int | v >= 0 } @-}
+type Nat = Int
+
+data T = T { getT :: Nat }
+
+{-@ f :: { t : T | getT t <= 0 } -> Nat @-}
+f :: T -> Nat
+f (T x) = x
```

Comment:
If `x` is a `Nat`, as the type of `getT` says, shouldn't this function be accepted by LH? 
If I try the following program, supposedly equivalent to the this test, LH accepts it.
```Haskell
{-@ LIQUID "--exact-data-cons" @-}
module AutoliftedFields00 where

{-@ type Nat = { v : Int | v >= 0 } @-}
type Nat = Int

{-@ data T = T { getT :: Nat } @-}
data T = T { getT :: Nat }

{-@ f :: { t : T | getT t <= 0 } -> Nat @-}
f :: T -> Nat
f (T x) = x
```

Reply:
I can confirm your program is accepted while the one I wrote is not... though I'm not sure why!

I do not understand fully why the one with the explicit data decl is accepted.  My understanding is that if `getT :: Nat` and `getT t <= 0` then the only available value for the function to return is when `x == 0`.  I suppose that subtypes `Nat`, and that's why it's accepted, since there is no "true" (in the HM sense) refinement type for anything?

Then it must be that `Nat` is being used as `Int` in the no decl case... No, I still don't understand.  Since, `getT t <= 0` would imply `x <= 0` which subtypes `Int`, just like in the above case!  I will investigate further.

Reply:
I understand it this way. LH needs to prove that `f` returns a `Nat`, since that is what the specification says. There are two hypothesis to accomplish this
1. the returned `x` is a `Nat` because `t = T x` (inferred from the pattern matching), and `T :: Nat -> T`.
2. `x = getT t` && `getT t <= 0`

So LH completes the proof using hypothesis (1).

Reply:
I can confirm that the reason is that the fields are being inferred to have (refinement) type `GHC.Types.Int` instead of `AutoliftedFields00.Nat`.  With the specification, the type is correctly `Nat`.  @nikivazou, do you happen to have an idea of why this would be?  I'm vaguely aware of expansion being done using the `Expand` typeclass, and I suspect it's being done too early on for fields/data decls, so that the alias is being expanded before LH has a chance to see it and know that it should use the corresponding LH alias.  However, the details are much more complicated than what I've been able to figure out myself.  For example, I don't know when the type for the field is first grabbed from GHC; if you knew this or could point me in the right direction it would be greatly appreciated!